### ISSUE TITLE
why I don't get the same output with the same input and same model?

### ISSUE THREAD

**Issue by masouduut94** [mvirgo/MLND-Capstone#31]
I use this code to get output file which is a road with detected path according to detected lines:

from os.path import splitext, basename

import numpy as np
import cv2

from moviepy.editor import VideoFileClip
from keras.models import load_model

model = load_model('full_CNN_model.h5')

class Lanes():
    def __init__(self):
        self.recent_fit = []
        self.avg_fit = []

lanes = Lanes()
vid_in = "line_1.mp4"
vid_output = 'line_1_out.mp4'

video_capture = cv2.VideoCapture()
fourcc = cv2.VideoWriter_fourcc(*'XVID')
name, extension = splitext(basename(vid_in))
output = name + '.avi'

if video_capture.open(vid_in):
    property_id = int(cv2.CAP_PROP_FRAME_COUNT)
    total_frames = int(cv2.VideoCapture.get(video_capture, property_id))
    frame_no = 1
    width, height = int(video_capture.get(cv2.CAP_PROP_FRAME_WIDTH)), \
                    int(video_capture.get(cv2.CAP_PROP_FRAME_HEIGHT))
    fps = video_capture.get(cv2.CAP_PROP_FPS)

    video_writer = cv2.VideoWriter(output, fourcc, fps, (width, height))
    while video_capture.isOpened():
        ret, frame = video_capture.read()
        if not ret:
            break
        
        small_img = np.array(np.resize(frame, (80, 160, 3)))
        small_img = small_img[None,:,:,:]
        
        prediction = model.predict(small_img)[0] * 255

        lanes.recent_fit.append(prediction)
        if len(lanes.recent_fit) > 5:
            lanes.recent_fit = lanes.recent_fit[1:]

        lanes.avg_fit = np.mean(np.array([i for i in lanes.recent_fit]), axis = 0)

        blanks = np.zeros_like(lanes.avg_fit).astype(np.uint8)
        lane_drawn = np.dstack((blanks, lanes.avg_fit, blanks))
        lane_image = np.resize(lane_drawn, (720, 1280, 3))

        result = cv2.addWeighted(lane_image, 1, lane_image, 1, 0)

        if frame_no % 20 == 0:
            print('# {}/{} frames processed!'.format(frame_no, total_frames))

        frame_no += 1
        video_writer.write(frame)
    video_capture.release()
    video_writer.release()

I also used the code in the original repository, but I did not get the same output in repository, Although I use the same steps.

Do I have to train another model now?

**Comment by mvirgo** [comment]
My apologies for not getting back to you earlier on this.

It might be a little easier to debug with the output video, but it could be a difference in how scipy's `imresize` function I used originally and the `np.resize` function you used do their re-sizing. Most likely, the input images once re-sized are no longer the same.

The second possibility is that there may be a difference in image color channels. OpenCV uses BGR, and since you are using it as the video capture, that may differ from how I trained on RGB images (and loaded with moviepy's `VideoFileClip`). That actually seems like the more likely issue, if that's actually occurring.
